### DISCUSSION TITLE
Block navigation without showing the browser confirm dialog.

### DISCUSSION THREAD

**Discussion by oc-jgm** [TanStack/router#779] [Q&A]
Does anyone know if there is a way to block navigation without showing the browser's default confirm dialog? I would like to show a custom modal within my app instead of the browser's default one.

**Comment by salemlf** [comment]
I'm also wondering this, according to the [current docs](https://tanstack.com/router/v1/docs/guide/navigation-blocking) it looks like the only customizable part is the message :( hope they add this feature as it's the only thing preventing me from switching over from React Router v6, I use [react-router-prompt](https://www.npmjs.com/package/react-router-prompt) rn with that.

Update: it looks like React Router just uses the history API behind the scenes for blocking/prompting before leaving a page. [Tanstack Router also appears to use this API.](https://github.com/TanStack/router/blob/beta/packages/react-router/src/react.tsx#L969) So _theoretically possible_ to implement something yourself using `history.block` and can provide your own dialog. Not an ideal solution, but may try it out and see if it can be used as a workaround, or open a PR if it's feasible for Tanstack. [Here's a guide](https://github.com/remix-run/history/blob/main/docs/blocking-transitions.md) on blocking page transitions using the history API
